x/ccv/consumer/keeper/throttle_retry.go

```diff
@@ -0,0 +1,109 @@
+package keeper
+
+import (
+	"fmt"
+	"time"
+
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+
+	consumertypes "github.com/cosmos/interchain-security/v3/x/ccv/consumer/types"
+)
+
+//
+// Throttling with retries follows a FSM design:
```

Comment:
nit: Might help if this comment-description started with what the states of the state machine are and then describe the actions in each state. E.g., we have 2 states "No Slash" and "Standby" and in case we're ("No Slash", receive slash packet) => ("Standby"), ...

Reply:
Good idea, I should draw up the state machine as well

Reply:
https://github.com/cosmos/interchain-security/commit/a834a5bf668c396b2bcd4fb0de692184d754efff